web/src/lib/utils/auth.ts

```diff
@@ -1,53 +1,61 @@
 import { api } from '@api';
 import { redirect } from '@sveltejs/kit';
 import { AppRoute } from '../constants';
-import { getSavedUser, setUser } from '$lib/stores/user.store';
+import { get } from 'svelte/store';
 import { serverInfo } from '$lib/stores/server-info.store';
+import { browser } from '$app/environment';
+import { user } from '$lib/stores/user.store';
 
 export interface AuthOptions {
   admin?: true;
+  public?: true;
 }
 
-export const getAuthUser = async () => {
+export const loadUser = async () => {
   try {
-    const { data: user } = await api.userApi.getMyUserInfo();
-    return user;
+    if (!get(user) && hasAuthCookie()) {
+      const { data } = await api.userApi.getMyUserInfo();
+      user.set(data);
+    }
+    return get(user);
   } catch {
-    return null;
+    return undefined;
   }
 };
 
+const hasAuthCookie = (): boolean => {
+  if (browser) {
+    const cookies = document.cookie.split('; ');
+    for (const cookie of cookies) {
+      const [name] = cookie.split('=');
+      if (name === 'immich_is_authenticated') {
+        return true;
+      }
+    }
+  }
+  return false;
+};
+
 export const authenticate = async (options?: AuthOptions) => {
   options = options || {};
+  const user = await loadUser();
 
-  const savedUser = getSavedUser();
-  const user = savedUser || (await getAuthUser());
-
-  if (!user) {
-    redirect(302, AppRoute.AUTH_LOGIN);
+  if (options.public && !user) {
```

Comment:
why do you care if the user is defined or not if the route is public?

Reply:
it seems like you want:
1. if public return we're done with all the checks
2. if not user, redirect to login, this is an auth required route
3. if admin route and not an admin, redirect to /photos